### ISSUE TITLE
@EnableResourceServer creates a FilterChain that matches possible unwanted endpoints with Spring Security 4.0.3

### ISSUE THREAD

**Issue by jebeaudet** [spring-projects/spring-security-oauth#634]
In Spring Security 4.x, since https://github.com/spring-projects/spring-security/commit/fbf3672ecaf0497a52c557b7d3a05951be30c7ef, the method `HttpSecurity#requestMatchers()`, doesn't override the previous invocations of `requestMatchers()`. 

When using `@EnableResourceServer` in the same application of an authorization server configured with `@EnableAuthorizationServer`, the imported class `ResourceServerConfiguration` creates a `FilterChain` mapping on the matcher `ResourceServerConfiguration#NotOAuthRequestMatcher`. This will basically match everything except the authorization endpoints. The only way to override this now is to call the `HttpSecurity.requestMatcher(RequestMatcher requestMatcher)` with a built `RequestMatcher`. This is IMO less fluent than user the `RequestMatcherConfigurer`. 

This issue is also present in the sparkl sample app. By enabling `debug=true` in `@EnableWebSecurity`, getting the login page yields to a different `FilterChain` being applied to the request : 
```
Request received for GET '/login.jsp':
org.apache.catalina.connector.RequestFacade@38363d8b
servletPath:/login.jsp
[...]

Security filter chain: [
  WebAsyncManagerIntegrationFilter
  SecurityContextPersistenceFilter
  HeaderWriterFilter
  LogoutFilter
  OAuth2AuthenticationProcessingFilter
  RequestCacheAwareFilter
  SecurityContextHolderAwareRequestFilter
  AnonymousAuthenticationFilter
  SessionManagementFilter
  ExceptionTranslationFilter
  FilterSecurityInterceptor
]
```
With Spring security 3.2.8 :
```
Request received for GET '/login.jsp':
org.apache.catalina.connector.RequestFacade@467a6273
servletPath:/login.jsp
[...]


Security filter chain: [
  WebAsyncManagerIntegrationFilter
  SecurityContextPersistenceFilter
  HeaderWriterFilter
  LogoutFilter
  UsernamePasswordAuthenticationFilter
  RequestCacheAwareFilter
  SecurityContextHolderAwareRequestFilter
  AnonymousAuthenticationFilter
  SessionManagementFilter
  ExceptionTranslationFilter
  FilterSecurityInterceptor
]
```
I'm thinking that maybe this part is superfluous in `ResourceServerConfiguration` : 
```
if (endpoints != null) {
	// Assume we are in an Authorization Server
	requests.requestMatchers(new NotOAuthRequestMatcher(endpoints
			.oauth2EndpointHandlerMapping()));
}
```
Opinions? Thanks!!

**Comment by dsyer** [comment]
I already noticed that, actually. I think that just using `requestMatcher()` instead of `requestMatchers()` is enough to recover the old behaviour in both Spring Security 3.2 and 4.0.

**Comment by dsyer** [comment]
Fixed in 21fce68a
